pallets/nfts/src/features/approvals.rs

```diff
@@ -148,15 +155,27 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	///   performed.
 	/// - `collection`: The collection ID containing the item.
 	/// - `item`: The item ID for which transfer approvals will be cleared.
+	/// - `witness_approvals`: Information on the collection approvals cleared. This must be
+	///   correct.
 	pub(crate) fn do_clear_all_transfer_approvals(
 		maybe_check_origin: Option<T::AccountId>,
 		collection: T::CollectionId,
 		item: T::ItemId,
+		witness_approvals: u32,
 	) -> DispatchResult {
 		let mut details =
 			Item::<T, I>::get(collection, item).ok_or(Error::<T, I>::UnknownCollection)?;
 
+		ensure!(details.approvals.len() as u32 == witness_approvals, Error::<T, I>::BadWitness);
```

Comment:
Additional code to support the witness approvals to `do_clear_all_transfer_approvals`.

Reply:
Why do we need this?

Reply:
I was not sure about this code to be honest. We can remove this if you think it is not needed.